test/ex_doc/formatter/html/templates_test.exs

```diff
@@ -52,6 +52,25 @@ defmodule ExDoc.Formatter.HTML.TemplatesTest do
     assert Templates.header_to_id("Git Options (<code class=\"inline\">:git</code>)") == "git-options-git"
   end
 
+  test "link headers" do
+    assert Templates.link_headings("<h2>Foo</h2><h2>Bar</h2>") == "<h2 id=\"foo\" class=\"section-heading\">\n  <a href=\"#foo\" class=\"hover-link\"><i class=\"icon-link\"></i></a>\n  Foo\n</h2>\n<h2 id=\"bar\" class=\"section-heading\">\n  <a href=\"#bar\" class=\"hover-link\"><i class=\"icon-link\"></i></a>\n  Bar\n</h2>\n"
```

Comment:
Sure, is there a preference between one or the other? If not I'll probably do the latter so it's easier to read.